syncstorage-spanner/src/stream.rs

```diff
@@ -144,19 +144,19 @@ where
     }
 
     // We could implement Stream::poll_next instead of this, but
-    // this is easier for now and we can refactor into the trait later
-    pub async fn next_async(&mut self) -> Option<DbResult<Vec<Value>>> {
+    // this is easier for now and we can refactor into the trait later.
+    pub async fn next_async(&mut self) -> DbResult<Option<Vec<Value>>> {
```

Comment:
Let's also rename this to more closely mimic [TryStream's naming](https://docs.rs/futures/latest/futures/prelude/stream/trait.TryStreamExt.html#method.try_next)
```suggestion
    pub async fn try_next_async(&mut self) -> DbResult<Option<Vec<Value>>> {
```